[WIP] Initial project structure

bushshrub opened the pull request:
Closes: #1
Closes: #2
Closes: #5 
Closes: #6 
Closes: #7
Closes: #8
Closes: #9
Closes: #10

TODO:

- [x] #5 
- [x] #6 
- [x] #7 
- [x] #8 
- [x] #9 
- [x] #10

cc @sudomaze

bushshrub committed:
Add basic gitignore

bushshrub committed:
Create LICENSE

bushshrub committed:
Ignore IDE-generated directories

bushshrub committed:
Create issue templates for bugs and features

bushshrub committed:
Create documentation directory

bushshrub committed:
Initial contributing guidelines and templates

bushshrub committed:
Create the base package and directory structure

bushshrub committed:
Create initial pyproject.toml

bushshrub committed:
Create CI workflow

bushshrub committed:
Create lint workflow

bushshrub committed:
Add mypy check to lint workflow

bushshrub committed:
Rename lint workflow's name to Linting

bushshrub committed:
Create some example scripts

bushshrub committed:
Some initial frameworks for the library

bushshrub committed:
Add CI to build docs

bushshrub committed:
Setup mkdocs with required plugins

bushshrub committed:
Expose public API in __init__

bushshrub committed:
Update mnist_torch example to demonstrate basic usage of our API

bushshrub committed:
Depend on pycln for dev

bushshrub committed:
Update our contributing guidelines

bushshrub committed:
Add example notebook

bushshrub committed:
Reformat code

bushshrub commented on mkdocs.yml:
```diff
@@ -0,0 +1,10 @@
+site_name: approx
```
Do we think mkdocs is the way to go? AFAIK, it's far easier to use as compared to sphinx.

ma7dev replied:
Yeah, let's use `mkdocs`. I have had my bad memories dealing with `sphinx` 😢

bushshrub replied:
Aight

bushshrub commented on approx/core/api.py:
```diff
@@ -0,0 +1,27 @@
+    Returns:
+        None
+    """
+    auto_select_backend()
```
`auto_select_backend()` returns a BackendEngine but it is not being used?

bushshrub commented on approx/core/backends/pytorch.py:
```diff
@@ -0,0 +1,7 @@
+
+
+class PyTorchBackend(BackendEngine):
+    pass
```
Should be implemented

bushshrub committed:
Fix api expose

ma7dev committed:
feat(core): Refactored core codebase & added tests

- Refactored codebase to be able to be functional
- Added tests to test the functionality of core codebase
- Included pre-commit for auto formatting
- Update pyproject.toml by moving torch and numpy to dev-dep

ma7dev committed:
feat(format) Formatting all files

ma7dev reviewed (approved):
<img src="https://media2.giphy.com/media/Q66ZEIpjEQddUOOKGW/giphy.gif"/>
